jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketEditor/S3TreeTable.kt

```diff
@@ -65,4 +85,51 @@ open class S3TreeTable(private val treeTableModel: S3TreeTableModel) : TreeTable
     fun invalidateLevel(node: S3TreeNode) {
         node.parent?.removeAllChildren()
     }
+
+    private fun createDropTarget(): DropTarget {
+        val dropTarget = DropTarget()
+        try {
+            dropTarget.addDropTargetListener(object : DropTargetAdapter() {
+                override fun drop(dropEvent: DropTargetDropEvent) {
+                    try {
+                        dropEvent.acceptDrop(DnDConstants.ACTION_COPY_OR_MOVE)
+                        val data = dropEvent.transferable.getTransferData(DataFlavor.javaFileListFlavor) as List<File>
+                        val row = rowAtPoint(dropEvent.location).takeIf { it >= 0 } ?: return
+                        val node = getNodeForRow(row) ?: return
+
+                        val lfs = LocalFileSystem.getInstance()
+                        val virtualFiles = data.mapNotNull {
```

Comment:
Does this mean I can do multiple files?

Reply:
Yep, box select and drag works as expected (minus the current UI of UploadObjectAction creating a box that blocks out the UI)